Android/src/org/droidplanner/android/fragments/geotag/GeoTagImagesService.java

```diff
@@ -0,0 +1,389 @@
+            geoTagTask.cancel(true);
+            geoTagTask = null;
+        }
+    }
+
+    private void geoTagImages(final Context context, File tlogFile) {
+        final String extMount = getExternalStorage(context);
+        if (extMount == null) {
+            return;
+        }
+
+        final ArrayList<File> photoFiles = new ArrayList<>();
+
+        List<File> photos = searchDir(extMount);
+        if (photos != null) {
+            photoFiles.addAll(photos);
+        }
+
+        if (photoFiles.size() == 0) {
+            sendFailedGeotaggingIntent("No photos on SD card for GoPro device.");
+            return;
+        }
+
+        Uri uri = Uri.fromFile(tlogFile);
+        Handler handler = new Handler();
+        TLogParser.getAllEventsAsync(handler, uri, new TLogParserFilter() {
+
+            @Override
+            public boolean includeEvent(TLogParser.Event event) {
+                return 180 == event.getMavLinkMessage().msgid;
+            }
+
+            @Override
+            public boolean shouldIterate() {
+                return true;
+            }
+
+
+        }, new TLogParserCallback() {
+            @Override
+            public void onResult(List<TLogParser.Event> eventList) {
+                if (eventList.size() < 0) {
+                    sendFailedGeotaggingIntent("No camera message events found");
+                    return;
+                }
+
+                if (geoTagTask != null) {
+                    geoTagTask.cancel(true);
+                }
+
+                File file = new File(getSaveRootDir(context), GEO_TAG_ROOT_NAME);
+
+                geoTagTask = new GeoTagTask(file, eventList, photoFiles);
+                geoTagTask.execute();
+            }
+
+            @Override
+            public void onFailed(Exception e) {
+                if (e instanceof NoSuchElementException) {
+                    sendFailedGeotaggingIntent("No camera message events found");
+                } else {
+                    sendFailedGeotaggingIntent(e.getMessage());
+                }
+            }
+        });
+    }
+
+    private String getExternalStorage(Context context) {
+        boolean hasNullFile = false;
+
+        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            File[] files = context.getExternalFilesDirs(null);
+            for (File extFile : files) {
+                if (extFile == null) {
+                    hasNullFile = true;
+                } else if (Environment.isExternalStorageRemovable(extFile)) {
+                    return findRootPath(extFile);
+                }
+            }
+        }
+        if (hasNullFile) {
+            sendFailedGeotaggingIntent("No external storage device found.");
+        } else {
+            sendFailedGeotaggingIntent("Incompatible device. No external SD card reader found.");
+        }
+        return null;
+    }
+
+    private static String findRootPath(File extFile) {
+        File currPath = null;
```

Comment:
@ne0fhyk Technically it's being checked before this method is called in getExternalStorage(). I check to see this device even has external storage. I can check here too if you want.